### PR TITLE
Support Numpy ints in the `torch.nn.functional.interpolate` dtype check

### PR DESCRIPTION
**Pull request by ezhang887**
In https://github.com/pytorch/pytorch/pull/99243, a check was added to ensure the `size` only contained integers.

This PR updates the check to also include numpy integers based on this comment (cc @kit1980): https://github.com/pytorch/pytorch/pull/99243#issuecomment-1646736646. Similar to the other commenter, I also ran into issues where existing software broke due to this after upgrading to PT2.1:

```
                if not torch.jit.is_scripting():
                    if not all(_is_integer(x) for x in size):
>                       raise TypeError(
                            "expected size to be one of int or Tuple[int] or Tuple[int, int] or "
                            f"Tuple[int, int, int], but got size with types {[type(x) for x in size]}"
                        )
E                       TypeError: expected size to be one of int or Tuple[int] or Tuple[int, int] or Tuple[int, int, int], but got size with types [<class 'numpy.int64'>, <class 'numpy.int64'>]

/conda-env/lib/python3.8/site-packages/torch/nn/functional.py:3924: TypeError
```

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -3776,10 +3781,12 @@ def upsample(input, size=None, scale_factor=None, mode="nearest", align_corners=
 
 def _is_integer(x) -> bool:
     r"""Type check the input number is an integer.
-    Will return True for int, SymInt and Tensors with integer elements.
+    Will return True for int, SymInt and Tensors / Numpy arrays with integer elements.
```
Comment (mikaylagawarecki) [inline torch/nn/functional.py:3784]: Hm to clarify the docstring it is not actually an numpy array but rather a numpy integer is that correct?
```suggestion
    Will return True for int, SymInt, Numpy integers and Tensors with integer elements.
```

**Review by mikaylagawarecki — APPROVED** [review]
Looks ok since the linked comment mentioned that interpolate used to work with size as a tuples of numpy integers on 2.0.1

File: torch/nn/functional.py (hunk 1 above)
Comment (ezhang887) [reply torch/nn/functional.py:3784]: Oops yes

**Pull request closed without merging**